runtime/src/asset.rs

```diff
@@ -1686,7 +1712,49 @@ impl<T: Trait> Module<T> {
         Ok(if general_status_code != ERC1400_TRANSFER_SUCCESS {
             general_status_code
         } else {
-            <percentage_tm::Module<T>>::verify_restriction(ticker, from_did, to_did, value)?
+            let mut final_result = true;
+            let mut is_valid = false;
+            let mut is_in_valid = false;
+            let mut force_valid = false;
+            let tms = Self::extensions((ticker, SmartExtensionType::TransferManager))
+                .into_iter()
+                .filter(|tm| Self::extension_details((ticker, tm)).is_archive == false)
+                .collect::<Vec<T::AccountId>>();
+            if tms.len() > 0 {
+                for tm in tms.into_iter() {
+                    let result = Self::verify_restriction(
+                        ticker,
+                        extension_caller.clone(),
+                        from_did,
+                        to_did,
+                        value,
+                        tm,
+                    );
+                    if result == RestrictionResult::Valid {
+                        is_valid = true;
+                    } else if result == RestrictionResult::Invalid {
+                        is_in_valid = true;
+                    } else if result == RestrictionResult::ForceValid {
+                        force_valid = true;
+                    }
+                }
+                //is_valid = force_valid ? true : (is_in_valid ? false : is_valid);
+                is_valid = if !force_valid {
+                    if is_in_valid {
+                        false
+                    } else {
+                        is_valid
+                    }
+                } else {
+                    true
+                };
+                final_result = is_valid;
+            }
+            if final_result {
+                return Ok(ERC1400_TRANSFER_SUCCESS);
+            } else {
+                return Ok(ERC1400_TRANSFER_FAILURE);
+            }
```

Comment:
It can be simplified to something like
```
if force_valid || !is_in_valid && is_valid {
    return Ok(ERC1400_TRANSFER_SUCCESS);
}
return Ok(ERC1400_TRANSFER_FAILURE);
```